lib/addons/keda/index.ts

```diff
@@ -0,0 +1,130 @@
+import { Construct } from 'constructs';
+import { ManagedPolicy } from "aws-cdk-lib/aws-iam";
+import { HelmAddOn, HelmAddOnUserProps, HelmAddOnProps } from "../helm-addon";
+import { ClusterInfo, Values } from "../../spi";
+import { setPath, createNamespace } from "../../utils";
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface KedaAddOnProps extends HelmAddOnUserProps {
+    /**
+     * Version of the helm chart to deploy
+     */    
+    version?: string,
+    /**
+     * Name of the KEDA operator
+     */
+    kedaOperatorName?: string,
+    /**
+     * Specifies whether a service account should be created. 
+     */
+    createServiceAccount?: boolean,
+    /**
+     * The name of the service account to use. If not set and create is true, a name is generated.
+     */
+    kedaServiceAccountName?: string,
+    /**
+     * securityContext: fsGroup
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */   
+    podSecurityContextFsGroup?: number,
+    /**
+     * securityContext:runAsUser
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */   
+    securityContextRunAsGroup?: number,
+    /**
+     * securityContext:runAsGroup
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */
+    securityContextRunAsUser?: number,
+    /**
+     * Th Dictionary of MAnaged IAM Roles which Sercice Account needs for IRSA Eg: irsaRoles: {"cloudwatch":"CloudWatchFullAccess", "sqs":"AmazonSQSFullAccess"}
+     */   
+    irsaRoles?: {}
+
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & KedaAddOnProps = {
+  name: "blueprints-keda-addon",
+  chart: "keda",
+  namespace:"keda",
+  version: "2.7.1",
+  release: "keda",
+  repository:  "https://kedacore.github.io/charts",
+  values: {},
+  createServiceAccount: true,
+  kedaOperatorName: "keda-operator",
+  kedaServiceAccountName: "keda-operator",
+  irsaRoles: {"cloudwatch":"CloudWatchFullAccess"}
+};
+
+/**
+ * Main class to instantiate the Helm chart
+ */
+export class KedaAddOn extends HelmAddOn {
+
+  readonly options: KedaAddOnProps;
+  constructor(props?: KedaAddOnProps) {
+    super({...defaultProps, ...props});
+    this.options = this.props as KedaAddOnProps;
+  }
+
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    //Create Service Account with IRSA
+    const cluster = clusterInfo.cluster;
+    let values: Values = populateValues(this.options);
+    if(this.options.createServiceAccount === false) {
```

Comment:
@shapirov103 Its intentional, If specified true, KEDA will create service account. In case user doesnt want to KEDA to create Service account, make it false and in this Case CDK will create Service account with IRSA. This is required if we want keda to consume metrics from AWS services like SQS and Kinesis
I will update comment int the interface inorder to avoid confusion

    /**
     * Specifies whether a service account should be created by keda. If provided false, CDK will create Service Account wth IAM Roles (IRSA). 
     */
    createServiceAccount?: boolean,